File: January/Jan_11/factorial.js
function factorial(number) {
    if (number === 0 || number === 1) {
      return 1;
    } else {
      let result = 1;
      for (let i = 2; i <= number; i++) {
        result *= i;
      }
      return result;
    }
  }
  
  // Test cases
  console.log(factorial(5)); // Should print 120
  console.log(factorial(0)); // Should print 1
  console.log(factorial(10)); // Should print 3628800
  

File: January/Jan_24/factorial.swift
import Foundation

func factorial(_ n: Int) -> Int {
    if n == 0 {
        return 1
    } else {
        return n * factorial(n - 1)
    }
}

func main() {
    print("Enter a number to calculate its factorial:")
    if let input = readLine(),
       let number = Int(input) {
        let result = factorial(number)
        print("The factorial of \(number) is \(result)")
    } else {
        print("Invalid input. Please enter a valid number.")
    }
}

main()
